pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/DeadLetterPolicy.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.policies.data;
+
+public class DeadLetterPolicy {
+
+    private final int maxRedeliverCount;
+
+    private String deadLetterTopic;
+
+
+    public DeadLetterPolicy(int maxRedeliverCount) {
+        this.maxRedeliverCount = maxRedeliverCount;
+    }
+
+    public DeadLetterPolicy(int maxRedeliverCount, String deadLetterTopic) {
```

Comment:
Instead of providing constructor overloads, we could use interface and builder approach like in other parts of the API. That way the code will long more descriptive. 

Eg: 
```java
/// This 
DeadLetterPolicy.builder()
   .maxRedeliverCount(10)
   .deadLetterTopic("my-dql-topic")
   .build()

// Instead of 
new DeadLetterPolicy(10, "my-dql-topic")
```

Reply:
Ok